### ISSUE TITLE
Memory State Persistence

### ISSUE THREAD

**Issue by kiplangatkorir** [GraphFusion/GraphFusion-NMN#16]
The `MemoryCell` class reinitializes hidden and cell states with zeros for each forward pass, causing potential loss of long-term memory context.

**Current Behavior**
```
hidden = torch.zeros(1, batch_size, self.hidden_size, device=input_data.device)
cell = torch.zeros(1, batch_size, self.hidden_size, device=input_data.device)
```
**Expected Behavior**

Maintain persistent memory states between forward passes
Implement proper state management for LSTM hidden states
Add mechanism to selectively reset states when needed

**Tasks**

- [x]  Implement state persistence mechanism
- [x]  Add state management utilities
- [x]  Create tests for state consistency
- [x]  Document state management approach
